staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -4528,6 +4536,31 @@ type PodSchedulingGate struct {
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 }
 
+// WorkloadReference identifies the Workload object and PodGroup membership
+// that a Pod belongs to. The scheduler uses this information to apply
+// workload-aware scheduling semantics.
+type WorkloadReference struct {
+	// Name defines the name of the Workload object this pod belongs to.
+	// Workload must be in the same namespace as the pod.
+	//
+	// +required
+	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
+
+	// PodGroup is the name of the PodGroup within the Workload that this pod
+	// belongs to.
+	//
+	// +required
+	PodGroup string `json:"podGroup" protobuf:"bytes,2,opt,name=podGroup"`
+
+	// PodGroupReplicaIndex specifies the replica index of the PodGroup that this
+	// pod belongs to. It is only set when the Workload uses a replicated
+	// scheduling mode. This index differentiates pods belonging to different replicas
+	// of the same PodGroup. When set, it must be a DNS label.
+	//
+	// +optional
+	PodGroupReplicaIndex string `json:"podGroupReplicaIndex,omitempty" protobuf:"bytes,3,opt,name=podGroupReplicaIndex"`
```

Comment:
any reason you want to use `string`? we typically use the pointer for optional values

```suggestion
	PodGroupReplicaIndex *string `json:"podGroupReplicaIndex,omitempty" protobuf:"bytes,3,opt,name=podGroupReplicaIndex"`
```

Reply:
We don't need to use a pointer, because empty replica index (key) is not an accepted value for this key. So, we wouldn't gain anything by adding additional nil checks.